include/aws/common/byte_buf.h

```diff
@@ -451,9 +451,16 @@ AWS_EXTERN_C_END
 AWS_STATIC_IMPL struct aws_byte_buf aws_byte_buf_from_c_str(const char *c_str) {
     struct aws_byte_buf buf;
     buf.buffer = (uint8_t *)c_str;
-    buf.len = strlen(c_str);
-    buf.capacity = buf.len;
+    if (!buf.buffer) {
+        buf.len = 0;
+        buf.capacity = 0;
+    } else {
+        buf.len = strlen(c_str);
+        /* considering the byte for the terminating null character */
+        buf.capacity = buf.len + 1;
```

Comment:
I worry about this, because having the capacity be higher than len implies that there's a byte at the end that's writable, but that's probably not safe, especially in cases where the cstr is a literal in the .data section. Can we instead not add 1 and just pretend the null doesn't exist?

Reply:
In that case, if `c_str` is an empty string, i.e., `strlen(c_str) == 0`, then the correct behavior would be to set `buf.buffer == NULL`? Or should I update `aws_byte_buf_is_valid` to consider valid either `buffer == NULL` or `strlen(buffer) == 0` when `len == 0`  and `capacity == 0`.

Reply:
I think that behavior here is fine, I'm just suggesting that you remove the `+ 1` from the capacity.